crate/operator/create.py

```diff
@@ -845,6 +845,8 @@ def get_discovery_service(
             owner_references=owner_references,
         ),
         spec=V1ServiceSpec(
+            # Headless service
+            cluster_ip="None",
```

Comment:
A quick explanation of what is happening here:

Setting the `cluster_ip` explcitly to "None" (the String `"None"`, not the null `None`) will mean that kubernetes will not create an internal load balancer for it. Instead, when any pods try to access this service, the DNS record for it will return multiple IP addresses - the IPs of the individual pods that this service is "fronting". 

As far as crate is concerned, this is great for service discovery - it means all the crate nodes immediately see all the other nodes, instead of randomly connecting via a load balancer and by chance ending up on one they don't know (yet).